src/hu_stuff.c

```diff
@@ -857,32 +902,76 @@ static void HU_widget_build_armor (void)
   // build the numeric amount init string
   M_snprintf(hud_armorstr + i, sizeof(hud_armorstr), "%3d", st_armor);
 
-  // color of armor depends on type
-  if (hud_armor_type)
+  const crange_idx_e cr = CRByArmor();
+  w_armor.cr = colrngs[cr];
+
+  // transfer the init string to the widget
+  HUlib_add_string_to_cur_line(&w_armor, hud_armorstr);
+}
+
+static void HU_widget_build_compact (void)
+{
+  char hud_compactstr[HU_MAXLINELENGTH];
```

Comment:
It never happens that two widget strings are built at the same time. At this point, would it make sense to use a common static global string buffer for all widgets?

Reply:
Sure, if it's more convenient. We need to make sure that this global buffer always has `0` at the end, right?